src/requestLib.js

```diff
@@ -13,32 +12,36 @@ var RequestLib = {
    * @param {http.IncomingMessage} clientRequest 
    * @param {Number} retries 
    */
-  _makeRequest: function (params, clientRequest, retries) {
+  _makeRequest: function (schemeObj, params, clientRequest, retries) {
     return new Promise(function (resolve, reject) {
-      var requestOptions = Object.assign({}, params.furtherRequestOptions, {
+      var requestOptions = {
+        headers:{}
+      };
+      // Initialize the request to be fired on behalf of the client
+      var request = null;
+      var keepAliveAgent = null;
+      keepAliveAgent = new schemeObj.Agent({keepAlive: true});  
```

Comment:
no need to create the agent again and again... can be selected conditionally wherever required...